build-system/tasks/e2e/describes-e2e.js

```diff
@@ -62,8 +79,24 @@ function describeEnv(factory) {
         fixtures.push(fixture);
       }
     });
+
+    const environments = spec.environments || defaultEnvironments;
+    const variants = Object.create(null);
```

Comment:
Feel free to push back, but I prefer not to use `reduce` outside of data-processing oriented code, since at least for me they always take longer to grok later than the equivalent simple loop pattern.

Reply:
Thats fine, I just prefer to limit the scope where values change. I think if you do write a lot of functional code, it can become easy to follow, but no need to change if you aren't comfortable with it.